DevelopmentProposals/reporterColumnLabels/reporterColumnLabels.md

```diff
@@ -54,14 +54,15 @@ and forcing the user to specify an annotation for each Output may not be practic
 - The TableReporter should use a default column name that is very likely to be meaningful.
 
 ### Architecture
-The selected design involves four main changes:
+The selected design involves six main changes:
 
 1. Rename "annotation" to "alias" (e.g., `AbstractInput::getAnnotation()` becomes `AbstractInput::getAlias()`).
 2. If the user provides the optional second argument when connecting an Output to an Input, store the string as the Input's alias; otherwise, the alias remains null.
-3. Create `getShortLabel()` and `getLongLabel()` methods on Input:
-  - `getShortLabel()` will be called by ConsoleReporter and will return `isNull(alias) ? output_name : alias`.
-  - `getLongLabel()` will be called by TableReporter and will return `isNull(alias) ? full_path_name : alias`.
-4. Create `setAlias()` methods analogous to the current `getAnnotation()` methods so the user can change the alias after (but not before) connecting.
+3. Create `getReportedLabel()` on Input, which will always return the alias if one is provided:
+  - `getReportedLabel()` will be called by a Reporter and will return `isNull(alias) ? full_path_name : alias`.
```

Comment:
The previous version said:
```.md
3. Create `getShortLabel()` and `getLongLabel()` methods on Input:
  - `getShortLabel()` will be called by ConsoleReporter and will return `isNull(alias) ? output_name : alias`.
  - `getLongLabel()` will be called by TableReporter and will return `isNull(alias) ? full_path_name : alias`.
```
That sounds specific to reporting. 

I think having four methods: `Input::` ` getName()`, `getAlias()`, `getLongLabel()`, `getShortLabel()` is confusing. I believe having one *label* method primarily associated with the act of reporting would be easier to understand, especially since `getLongLabel()` and `getShortLabel()` yield the same value whenever there is an alias. IMHO `getDescriptiveLabel()` will not be any more descriptive than the alias, but I am fine with being outvoted for the name. The more important design aspect, for me, is that we do not introduce two methods where one can suffice.